README.md

```diff
@@ -13,3 +13,25 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+
+Part 1:
+1. An implementation may have a average case that differs from that of the same implementation, so you cannot generalize that all implementations have the same best, average or worst case.
+2. Depending on how you implement the given data structure, this can cause a different analysis from the asymptotic analysis described in the beginning.
+3. Depending on the machines the code runs on, the same code can vary in time. Asymptotic analysis does not take in external variables.
+
+Part 2:
+Within binary trees, they usually have a complexity of log2(n). So, to find the estimated runtime we can find this by inputting our values. For 1000 elements the runtime is estimated at log2(1000) = 9.9 seconds. For 10000 elements, the runtime is estimated at log2(10000) = 13.3 seconds. This is just an estimation. To see how the logarithmic graph grows to these two points, we can divide them to see how much they grow, (13.3/9.9) = 1.34. So, now taking into account the runtime we are given for 1000 elements which is 5 seconds, we can multiply our rate that goes from 1000 to 10000 by the number resulting in 5 * 1.34 = 6.7 seconds. 
+
+Part 3:
+1. Depending on how far the element is down, and if the tree is very unbalanced so it resembles more of a linked list, this may make the runtime longer.
```

Comment:
In the worst case, it would be linear -- could that account for a difference this large?